File: examples/Hello.java
// Java — Example for Hypervisor VM
// Run: docker_run.bat examples\Hello.java

public class Hello {
    public static void main(String[] args) {
        System.out.println("Hello from Java on Linux! 🐧");
        System.out.println("Running inside Docker Hypervisor!\n");

        // Factorial
        System.out.println("Factorials:");
        for (int i = 1; i <= 10; i++) {
            System.out.printf("  %2d! = %d%n", i, factorial(i));
        }
    }

    static long factorial(int n) {
        long result = 1;
        for (int i = 2; i <= n; i++) result *= i;
        return result;
    }
}

File: examples/hello.js
// JavaScript (Node.js) — Example for Hypervisor VM
// Run: docker_run.bat examples\hello.js

console.log("Hello from JavaScript (Node.js) on Linux! 🐧");
console.log("Running inside Docker Hypervisor!\n");

// Async example
const delay = ms => new Promise(resolve => setTimeout(resolve, ms));

async function main() {
    const languages = ["Custom VM", "Python", "C", "C++", "Java", "JavaScript"];
    console.log("Languages supported by HypervisorVM:");
    for (const lang of languages) {
        console.log(`  ✔ ${lang}`);
    }

    console.log("\nFibonacci (first 10):");
    let a = 0, b = 1;
    for (let i = 0; i < 10; i++) {
        process.stdout.write(a + " ");
        [a, b] = [b, a + b];
    }
    console.log();
}

main();
